internal/restarter/restart.go

```diff
@@ -0,0 +1,25 @@
+package restarter
+
+import (
+	"context"
+	operatorv1alpha2 "github.com/kyma-project/istio/operator/api/v1alpha2"
+	"github.com/kyma-project/istio/operator/internal/described_errors"
+)
+
+type Restarter interface {
+	Restart(ctx context.Context, istioCR *operatorv1alpha2.Istio) described_errors.DescribedError
```

Comment:
It would be could to document that `Restart` does not necessarily mean deterministic restart of the component, but an evaluation of the predicates and then a restart if needed

Reply:
I mean in package comment, not a full blown `md` docu entry